File: Data Structures & Algorithms/non-overlapping-intervals/submission-6.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        if(intervals.empty()) return 0;
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> ans;
        ans.push_back(intervals[0]);
        int count = 1;
        int i = 0;
        while(count < intervals.size()){
            int first = intervals[count][0];
            int max = ans.back()[1];
            if(first < max){
                if(intervals[count][1]<max){
                    ans.back() = intervals[count];
                }
                
                i++;
            }
            else{
                ans.push_back(intervals[count]);
                }
            count ++;
        }
        return i;
    }
};

File: Data Structures & Algorithms/is-subsequence/submission-6.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        lt, ls = len(t)-1, len(s)-1
        while lt >= 0:
            if ls < 0:
                return True
            if s[ls] == t[lt]:
                ls -=1
            lt -=1

        return ls <0
